models/repo.go

```diff
@@ -1472,6 +1472,8 @@ func DeleteRepository(doer *User, uid, repoID int64) error {
 		&LanguageStat{RepoID: repoID},
 		&Comment{RefRepoID: repoID},
 		&Task{RepoID: repoID},
+		&ProtectedBranch{RepoID: repoID},
```

Comment:
Hm, this should be a bug and we need a doctor to clean that.

Reply:
I will create a new PR with this change and a doctor command.